### DISCUSSION TITLE
Guideance needed

### DISCUSSION THREAD

**Discussion by PrashanthKumar0** [klange/toaruos#239] [General]
Decades of work . hats off to you sir .

It would be great if you provide some sites / reference / books that you learned from .

i would like to hear from you about these.

I am just beginning to understand these stuff so it would be really helpful for me and others too .

**Comment by klange** [comment]
> Decades of work . hats off to you sir .

Really just the one, so far ;)

> sites

https://wiki.osdev.org/

The [osdev.org forum](https://forum.osdev.org/), [/r/osdev](https://www.reddit.com/r/osdev), and `#osdev` on `irc.libera.chat` are all good places to ask question as a beginner. 

> books

Most books about OS dev are high-level theory and I haven't found any of them useful. The one actual book I've referenced was [TCP/IP Illustrated](https://www.oreilly.com/library/view/tcpip-illustrated-volume/9780132808200/).

**Reply by PrashanthKumar0** [reply]
Thank You so much :)
